### ISSUE TITLE
Running 'elm make' with compilation errors returns exit code 0

### ISSUE THREAD

**Issue by sashaafm** [elm/compiler#2115]
**Quick Summary:** When running `elm make` and the source code has compilation errors the exit code from the binary is 0 (zero) when it should be an integer > 1 since zero means success.

## SSCCE

```elm
module main exposing ()
```

```shell
elm make src/Main.elm
echo $?
```
- **Elm:** 0.19.1
- **Browser:** Not applicable
- **Operating System:** `Linux dev 4.19.76-linuxkit #1 SMP Fri Apr 3 15:53:26 UTC 2020 x86_64 GNU/Linux`

## Additional Details
Providing a correct exit code (POSIX compliant?) provides better semantics and user experience around the `elm` binary. My current use case is that doing something like `elm make src/Main.elm && digest_static_assets` does not work as expected. In this use case I would expect errors in the compilation process to return exit code > 0 so that the `digest_static_assets` executable is not run.

**Comment by jfmengels** [comment]
@sashaafm I tried your example, and it seems to work for me (the exit code is `1` as expected). I have also been relying on this behavior for a long time and haven't run into it incorrectly not working before.

I am personally running Ubuntu 20.04 (but I have tried it on other versions of the OS).

Have you noticed this for specific compilation errors, specific projects or for your specific OS (I can't figure out what OS you have)?
